pkgs/applications/version-management/gitlab/update.py

```diff
@@ -155,30 +138,26 @@ def update_rubyenv():
     # load rev from data.json
     data = _get_data_json()
     rev = data['rev']
+    version = data['version']
 
-    gemfile = repo.get_file('Gemfile', rev)
-    gemfile_lock = repo.get_file('Gemfile.lock', rev)
-
-    with open(rubyenv_dir / 'Gemfile', 'w') as f:
-        f.write(re.sub(f'.*({"|".join(VENDORED_GEMS)}).*', "", gemfile))
+    for fn in ['Gemfile.lock', 'Gemfile']:
+        with open(rubyenv_dir / fn, 'w') as f:
+            f.write(repo.get_file(fn, rev))
 
-    with open(rubyenv_dir / 'Gemfile.lock', 'w') as f:
-        f.write(gemfile_lock)
+    # Fetch some vendored dependencies temporarily in order to build the gemset.nix
+    subprocess.check_output(['rm', '-rf', 'vendor/gems'], cwd=rubyenv_dir)
+    subprocess.check_output(['mkdir', '-p', 'vendor/gems'], cwd=rubyenv_dir)
+    subprocess.check_output(['sh', '-c', f'curl -L https://gitlab.com/gitlab-org/gitlab/-/archive/v{version}-ee/gitlab-v{version}-ee.tar.bz2?path=vendor/gems | tar -xj --strip-components=3'], cwd=f'{rubyenv_dir}/vendor/gems')
 
     subprocess.check_output(['bundle', 'lock'], cwd=rubyenv_dir)
     subprocess.check_output(['bundix'], cwd=rubyenv_dir)
 
-    with open(rubyenv_dir / 'Gemfile', 'w') as f:
-        for gem in VENDORED_GEMS:
-            gemfile = gemfile.replace(f'path: \'vendor/gems/{gem}\'', f'path: \'{gem}\'')
-
-        f.write(gemfile)
+    # Patch the gemset, but keep the original file so that bundix won't complain.
```

Comment:
Bundix is ran before this, though? I feel like it would be better to not have two copies of the gemset in-tree for no reason, unless I'm misunderstanding something.

Reply:
Good point. I have removed `rubyEnv/gemset-patched.nix` and have rewritten the updater a little:

```diff
diff --git a/pkgs/applications/version-management/gitlab/default.nix b/pkgs/applications/version-management/gitlab/default.nix
index 19d53dc57a7..e6cd605a51e 100644
--- a/pkgs/applications/version-management/gitlab/default.nix
+++ b/pkgs/applications/version-management/gitlab/default.nix
@@ -20,7 +20,7 @@ let
     inherit ruby;
     gemdir = ./rubyEnv;
     gemset =
-      let x = import (gemdir + "/gemset-patched.nix") src;
+      let x = import (gemdir + "/gemset.nix") src;
       in x // {
         gpgme = x.gpgme // {
           nativeBuildInputs = [ pkg-config ];
diff --git a/pkgs/applications/version-management/gitlab/update.py b/pkgs/applications/version-management/gitlab/update.py
index a7bfc60f668..266ca3e4ceb 100755
--- a/pkgs/applications/version-management/gitlab/update.py
+++ b/pkgs/applications/version-management/gitlab/update.py
@@ -144,18 +144,17 @@ def update_rubyenv():
         with open(rubyenv_dir / fn, 'w') as f:
             f.write(repo.get_file(fn, rev))
 
-    # Fetch some vendored dependencies temporarily in order to build the gemset.nix
-    subprocess.check_output(['rm', '-rf', 'vendor/gems'], cwd=rubyenv_dir)
+    # Fetch vendored dependencies temporarily in order to build the gemset.nix
     subprocess.check_output(['mkdir', '-p', 'vendor/gems'], cwd=rubyenv_dir)
     subprocess.check_output(['sh', '-c', f'curl -L https://gitlab.com/gitlab-org/gitlab/-/archive/v{version}-ee/gitlab-v{version}-ee.tar.bz2?path=vendor/gems | tar -xj --strip-components=3'], cwd=f'{rubyenv_dir}/vendor/gems')
 
+    # Undo our gemset.nix patches so that bundix runs through
+    subprocess.check_output(['sed', '-i', '-e', '1d', '-e', 's:\\${src}/::g' , 'gemset.nix'], cwd=rubyenv_dir)
+
     subprocess.check_output(['bundle', 'lock'], cwd=rubyenv_dir)
     subprocess.check_output(['bundix'], cwd=rubyenv_dir)
 
-    # Patch the gemset, but keep the original file so that bundix won't complain.
-    subprocess.check_output(['cp', 'gemset.nix', 'gemset-patched.nix'], cwd=rubyenv_dir)
-    subprocess.check_output(['sed', '-i', '-e', '1i\\src:\\n', '-e', 's:path = \\(vendor/[^;]*\\);:path = "${src}/\\1";:g', 'gemset-patched.nix'], cwd=rubyenv_dir)
-
+    subprocess.check_output(['sed', '-i', '-e', '1i\\src:', '-e', 's:path = \\(vendor/[^;]*\\);:path = "${src}/\\1";:g', 'gemset.nix'], cwd=rubyenv_dir)
     subprocess.check_output(['rm', '-rf', 'vendor'], cwd=rubyenv_dir)
 
 ```